docs/reference/search/request/highlighting.asciidoc

```diff
@@ -929,3 +943,165 @@ Response:
+fragments in some raw form, and then populate them with actual text.
+
+A highlighter uses `pre-tags`, `post-tags` to encode highlighted terms.
+
+
+===== An example of the work of the plain highlighter
+
+Let's look in more details how plain highlighter works.
+
+We index the following text into the `text` field of a
+document using `english` analyzer:
+
+    "For you I'm only a fox like a hundred thousand other foxes. But if you tame me, we'll need each other.
+    You'll be the only boy in the world for me. I'll be the only fox in the world for you."
+
+
+And we ran the following query with a highlight request:
+
+[source,js]
+--------------------------------------------------
+{
+    "query": {
+        "match_phrase" : {"text" : "only fox"}
+    },
+    "highlight": {
+        "type" : "plain",
+        "number_of_fragments" : 3,
+        "fields": {
+            "text": {"fragment_size": 50}
+        }
+    }
+}
+--------------------------------------------------
+// NOTCONSOLE
+
+Plain highlighter begins with analyzing the given text,
+using the specified `english` analyzer, and creates a token stream from it:
+
+    {"token":"you","start_offset":4,"end_offset":7,"position":1},
+    {"token":"i'm","start_offset":8,"end_offset":11,"position":2},
+    {"token":"onli","start_offset":12,"end_offset":16,"position":3},
+    {"token":"fox","start_offset":19,"end_offset":22,"position":5},
+    {"token":"like","start_offset":23,"end_offset":27,"position":6},
+    {"token":"hundr","start_offset":30,"end_offset":37,"position":8},
+    {"token":"thousand","start_offset":38,"end_offset":46,"position":9},
+    {"token":"other","start_offset":47,"end_offset":52,"position":10},
+    {"token":"fox","start_offset":53,"end_offset":58,"position":11},
+    {"token":"you","start_offset":67,"end_offset":70,"position":14},
+    {"token":"tame","start_offset":71,"end_offset":75,"position":15},
+    {"token":"me","start_offset":76,"end_offset":78,"position":16},
+    {"token":"we'll","start_offset":80,"end_offset":85,"position":17},
+    {"token":"need","start_offset":86,"end_offset":90,"position":18},
+    {"token":"each","start_offset":91,"end_offset":95,"position":19},
+    {"token":"other","start_offset":96,"end_offset":101,"position":20},
+    {"token":"you'll","start_offset":103,"end_offset":109,"position":21},
+    {"token":"onli","start_offset":117,"end_offset":121,"position":24},
+    {"token":"boi","start_offset":122,"end_offset":125,"position":25},
+    {"token":"world","start_offset":133,"end_offset":138,"position":28},
+    {"token":"me","start_offset":143,"end_offset":145,"position":30},
+    {"token":"i'll","start_offset":147,"end_offset":151,"position":31},
+    {"token":"onli","start_offset":159,"end_offset":163,"position":34},
+    {"token":"fox","start_offset":164,"end_offset":167,"position":35},
+    {"token":"world","start_offset":175,"end_offset":180,"position":38},
+    {"token":"you","start_offset":185,"end_offset":188,"position":40}
+
```

Comment:
The `unified` highlighter does not index all terms but only those that can match the query. This is an issue currently in the `plain` highlighter since it caches all these terms in memory so we shouldn't document this and rely on the `unified` highlighter instead.